addons/web/static/src/core/advanced_select/advanced_select.js

```diff
@@ -0,0 +1,123 @@
+/** @odoo-module **/
+
+import { Component, useState, useRef } from "@odoo/owl";
+import { Dropdown } from "@web/core/dropdown/dropdown";
+import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import { _lt } from "@web/core/l10n/translation";
+import { useDebounced } from "@web/core/utils/timing";
+import { scrollTo } from "@web/core/utils/scrolling";
+
+export class AdvancedSelect extends Component {
+    setup() {
+        this.nextGroupId = 0;
+        this.nextOptionId = 0;
+
+        this.state = useState({ groups: [] });
+        this.inputRef = useRef("inputRef");
+        this.debouncedOnInput = useDebounced(this.onInput, 250);
+    }
+
+    onOpened(scrollContainer) {
+        if (this.inputRef.el) {
+            this.inputRef.el.focus();
+        }
```

Comment:
I'm doing the focus manually because the useAutofocus does not work properly in this case. The dropdown seems to mess with the input element state/useEffect. I don't quite understand why but it only focuses when the dropdown is opened a second time.